ttnn/cpp/ttnn/operations/moreh/moreh_fold/device/fold_program_factory_rm.cpp

```diff
@@ -0,0 +1,217 @@
+    bool out_is_dram = output.buffer()->buffer_type() == tt::tt_metal::BufferType::DRAM ? 1 : 0;
+
+    const std::vector<uint32_t> reader_compile_time_args{
+        static_cast<uint32_t>(src_is_dram),
+        static_cast<uint32_t>(input_cb_index),
+        static_cast<uint32_t>(output_cb_index),
+    };
+
+    std::vector<uint32_t> writer_compile_time_args{
+        static_cast<uint32_t>(out_is_dram),
+        static_cast<uint32_t>(output_cb_index),
+    };
+
+    const auto reader_kernel_file = "ttnn/cpp/ttnn/operations/moreh/moreh_fold/device/kernels/reader_fold_rm.cpp";
+    const auto writer_kernel_file = "ttnn/cpp/ttnn/operations/moreh/moreh_fold/device/kernels/writer_fold_rm.cpp";
+
+    const auto reader_kernel_id = tt::operations::primary::CreateReadKernel(
+        program, reader_kernel_file, all_cores, reader_compile_time_args, reader_defines);
+    const auto writer_kernel_id = tt::operations::primary::CreateWriteKernel(
+        program, writer_kernel_file, all_cores, writer_compile_time_args, writer_defines);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      RuntimeArgs SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    const auto input_addr = input.buffer()->address();
+    const auto output_addr = output.buffer()->address();
+
+    uint32_t start_id = 0;
+    auto cores = grid_to_cores(num_cores, num_cores_x, num_cores_y, false);
+    uint32_t g1_numcores = core_group_1.num_cores();
+    for (uint32_t i = 0; i < cores.size(); ++i) {
+        const CoreCoord& core = cores.at(i);
+        uint32_t num_units_per_core = i < g1_numcores ? num_units_per_core_group_1 : num_units_per_core_group_2;
+
+        vector<uint32_t> reader_args = {
+            input.buffer()->address(),
+            N,
+            C,
+            H,
+            W,
+            kernel_size_h,
+            kernel_size_w,
+            stride_h,
+            stride_w,
+            padding_h,
+            padding_w,
+            dilation_h,
+            dilation_w,
+            LH,
+            LW,
+            aligned_input_cb_page_size,
+            aligned_output_cb_page_size,
+            start_id,
+            num_units_per_core};
+
+        vector<uint32_t> writer_args = {
+            output.buffer()->address(), aligned_output_cb_page_size, start_id, num_units_per_core};
+
+        SetRuntimeArgs(program, reader_kernel_id, core, reader_args);
+        SetRuntimeArgs(program, writer_kernel_id, core, writer_args);
+        start_id += num_units_per_core;
+    }
+
+    return {std::move(program), {reader_kernel_id, writer_kernel_id, cores}};
+}
+
+void MorehFoldOperation::ProgramFactory::override_runtime_arguments(
+    cached_program_t& cached_program,
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& tensor_return_value) {
```

Comment:
Name it "output" instead of tensor_return_value

Reply:
Resolved in last commit.